docs/developer-documentation.md

```diff
@@ -0,0 +1,192 @@
+**Zulip Terminal** is a light and fast terminal client for [Zulip](https://zulipchat.com). It's written in python and :snake: only.
```

Comment:
Re this first commit, I would definitely note in the commit title or description, if this is a straight port of the wiki data. I think you said that was correct, but it would be good to be explicit about this.

You could even mention how it was ported across, ie. we discussed the wiki git repo?